src/EventCalendar.js

```diff
@@ -626,6 +1014,117 @@ class EventCalendar {
+        }
+
+        // If we're using US localisation, dates will be formatted as MM/DD/YYYY in the UI.
+        if (locale.toLowerCase() === 'en'  || locale.toLowerCase() === 'en-us') {
+            _self.localeFormat = _self.dateFormatUS;
+        }
+    }
+
+    /**
+     * Checks the current format of `date` and reformats it as _self.localeFormat (probably YYYY-MM-DD) for internal
+     * use within the CLNDR.
+     * 
+     * @param {string} date The date to convert
+     */
+    internalDate (date) {
+        let _self = this;
+
+        if (date === 'Invalid date') {
+            return;
+        }
+
+        // If already internal format, just return the same value
+        if (moment(date, _self.dateFormatInternal, true).isValid()) {
+            return date;
+        }
+
+        // Return `today` if no date is provided
+        if (typeof date === 'undefined' || !date.length) {
+            return moment(_self.today, _self.dateFormatInternal).format(_self.localeFormat);
+        }
+        
+        // Otherwise, reformat the date
+        return moment(date, _self.localeFormat).format(_self.dateFormatInternal);
+    }
+
+    /**
+     * Pluralises a string given a quantity of 'things'.
+     * 
+     * @param {string} noun The word to pluralise
+     * @param {integer} quantity The quantity to determine whether to pluralise
+     */
+    pluralise (noun, quantity) {
+        return quantity > 1 ? noun + 's' : noun;
+    }
+
+    /**
+     * Check whether the date of the month for `date` is before `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
+    static isBeforeDate (dateToCompareTo, date) {
+        return date.date() < dateToCompareTo.date();
+    }
+
+    /**
+     * Check whether the date of the month for `date` is after `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
+    static isAfterDate (dateToCompareTo, date) {
+        return date.date() > dateToCompareTo.date();
+    }
+
+    /**
+     * Check whether `date` is in the same month as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
+    static isInMonth (dateToCompareTo, date) {
+        return date.month() === dateToCompareTo.month();
+    }
+    
+    /**
+     * Check whether `date` is in the same year as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
+    static isInYear (dateToCompareTo, date) {
+        return date.year() === dateToCompareTo.year();
+    }
+
+    /**
+     * Check whether two moment instances are not the same date
+     * 
+     * @param {moment} a Haystack
+     * @param {moment} b Needle
+     */
+    static doesNotMatchDate (a, b) {
+        return a.format(this.dateFormat) != b.format(this.dateFormat);
+    }
 }
 
 module.exports = EventCalendar;
```

Comment:
I think from a scale point of view we passed the point when this class should be split down into a bunch of smaller services a while ago. This file is a monster. There are some methods in here that could be their own classes. I think splitting this down will make it easier to test and work on in the future.